### PR TITLE
option to add a custom prefix to all of Tailwind's generated utility classes

### PR DESCRIPTION
**Pull request by daryledesilva**
example:
```
tailwindo path/to/directory/ --prefix=tw-
```

### CODE REVIEW

File: src/Converter.php [hunk 1]
```diff
@@ -19,6 +19,9 @@ class Converter
 
     protected $components = [];
 
+    /** @var string */
+    protected $prefix = '';
```
Comment (abdumu) [inline src/Converter.php]: If we are not allowing empty prefix, this could be a `null`

File: src/Converter.php [hunk 2]
```diff
@@ -71,6 +74,20 @@ public function setGenerateComponents(bool $value): self
...
+    /**
+     * The prefix option allows you to add a custom prefix to all of Tailwind's generated utility classes. This can be really useful when layering Tailwind on top of existing CSS where there might be naming conflicts.
+     *
+     * @param string $prefix
+     *
+     * @return Converter
+     */
+    public function setPrefix(string $prefix): self
+    {
+        $this->prefix = trim($prefix);
```
Comment (abdumu) [inline src/Converter.php]: `if(! empty($prefix))  ... { .. }` ?

File: src/Converter.php [hunk 3]
```diff
@@ -210,7 +227,16 @@ function ($match) use ($replace) {
                         $this->components[$match['given']] = preg_replace('/\{tailwindo\|([^\}]+)\}/', '$1', $replace);
                     }
 
-                    return $replace;
+                    return $this->prefix ? trim(implode(' ', array_filter(array_map(function ($class) {
```
Comment (abdumu) [inline src/Converter.php]: Use `if` conditions and multiple lines for readability. multiple functions in one line makes debugging harder.

**Review by abdumu — CHANGES_REQUESTED** [review]
This a great addition to the tool. I would love to see a test even if it's one line to make sure it's not affecting the output.

Thanks

**Comment by daryledesilva** [comment]
> This a great addition to the tool. I would love to see a test even if it's one line to make sure it's not affecting the output.
> 
> Thanks

thanks! sorry for the test, i was in a hurry yesterday 😅 please check latest updates and let me know if there's anything else that needs to be done!

**Comment by daryledesilva** [comment]
> This a great addition to the tool. I would love to see a test even if it's one line to make sure it's not affecting the output.
> 
> Thanks

@abdumu thanks! sorry for the test, i was in a hurry yesterday 😅 please check latest updates and let me know if there's anything else that needs to be done!

**Comment by abdumu** [comment]
thanks

**Comment by serwin35** [comment]
does it work on the official version?

"The "--prefix" option does not exist."

**Comment by daryledesilva** [comment]
> does it work on the official version?
> 
> "The "--prefix" option does not exist."

@serwin35 i believe it's not in the latest release yet https://github.com/awssat/tailwindo/compare/3.0.2...master

**Pull request merged**